plugins/woocommerce/src/Blocks/Utils/CartCheckoutUtils.php

```diff
@@ -44,6 +44,32 @@ public static function is_checkout_block_default() {
 		return $checkout_page_id && has_block( 'woocommerce/checkout', $checkout_page_id );
 	}
 
+
+	/**
+	 * Checks if the template overriding the page loads the page content or not.
+	 * Templates by default load the page content, but if that block is deleted the content can get out of sync with the one presented in the page editor.
+	 *
+	 * @param string $block The block to check.
+	 *
+	 * @return bool true if the template has out of sync content.
+	 */
+	public static function is_overriden_by_custom_template_content( string $block ): bool {
+
+		$block = str_replace( 'woocommerce/', '', $block );
+
+		if ( wc_current_theme_is_fse_theme() ) {
+			$templates_from_db = BlockTemplateUtils::get_block_templates_from_db( array( 'page-' . $block ) );
+			foreach ( $templates_from_db as $template ) {
+				if ( ! has_block( 'woocommerce/page-content-wrapper', $template->content ) ) {
```

Comment:
Why are just checking if the `page-content-wrapper` is present? What if the merchant changes the existing template from the editor? I tried changing the template by replacing the block with the classic checkout within the template Editor. It didn't update the status to `This page's content is overridden by custom template content.` Is this expected?

Reply:
Thanks for having a look Tarun ☺️
The `page-content-wrapper` loads content from the defined page, if you delete it from the template editing the page will only influence the cart/checkout URL.
Not sure if I'm fully understanding your question, do you think we should be doing additional checks?

>I tried changing the template by replacing the block with the classic checkout within the template Editor. It didn't update the status to This page's content is overridden by custom template content. Is this expected?

It will only show that if you delete the `page-content-wrapper`,  if you just change the content it won't display that message

<img width="349" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/d2b02b9e-8d5d-464c-9528-5e9187c903dc">

Reply:
Gotcha! Thank you for clarifying that. 

>Not sure if I'm fully understanding your question, do you think we should be doing additional checks?

I tried making the other changes, but it also updates the page. `page-content-wrapper` check would be enough. 👍